pkg/planner/core/task.go

```diff
@@ -861,7 +875,23 @@ func (p *NominalSort) Attach2Task(tasks ...base.Task) base.Task {
 	return t
 }
 
-func (p *PhysicalTopN) getPushedDownTopN(childPlan base.PhysicalPlan) *PhysicalTopN {
+func (p *PhysicalTopN) getPushedDownTopN(childPlan base.PhysicalPlan, storeTp kv.StoreType) (*PhysicalTopN, *PhysicalTopN) {
+	var newGlobalTopN *PhysicalTopN
+
+	fixValue := fixcontrol.GetBoolWithDefault(p.SCtx().GetSessionVars().GetOptimizerFixControlMap(), fixcontrol.Fix56318, true)
+	// HeavyFunctionOptimize: if TopN's ByItems is a HeavyFunction (currently mainly for Vector Search), we will change
+	// the ByItems in order to reuse the function result.
+	if fixValue && ContainHeavyFunction(p.ByItems[0].Expr) {
```

Comment:
We need to add a condition "compare the line size between add projection or not"
In some case, we add column in projection for order by column which will cause the network consume increasing. 
If too many columns are added or the column type is too large, the speed will be slowed down.